### ISSUE TITLE
Error: ID was missing the `serverFarms` element

### ISSUE THREAD

**Issue by hanyang1986** [hashicorp/terraform-provider-azurerm#24721]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.52

### AzureRM Provider Version

3.89

### Affected Resource(s)/Data Source(s)

azurerm_app_service_managed_certificate

### Terraform Configuration Files

```hcl
resource "azurerm_app_service_managed_certificate" "this" {
  custom_hostname_binding_id = azurerm_app_service_custom_hostname_binding.this.id
}
```

### Debug Output/Panic Output

```shell
Error: ID was missing the `serverFarms` element
```

### Expected Behaviour

The error appears after terraform apply.

**Comment by tombuildsstuff** [comment]
hey @hanyang1986 

Thanks for opening this issue here.

Taking a look through here it appears this is a duplicate of #24662 which has been fixed by #24664 and will ship in this weeks release - as a fix for this has already been merged I'm going to close this issue for the moment, but as this issue has been assigned to the milestone a comment will be posted here once the release is available.

To workaround this in the interim, you should be able to pin to an older version of the Provider (e.g. `=3.87.0`) using the `required_versions` block within the `terraform` block:

```terraform
terraform {
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "=3.87.0"
    }
  }
}
```

Thanks!

**Comment by EngZaz** [comment]
When I tried version 3.90.0 i got this error
Error: creating/updating App Service Managed Certificate "**.**.com" (Resource Group "****"): web.CertificatesClient#CreateOrUpdate: Failure sending request: StatusCode=409 -- Original Error: autorest/azure: Service returned an error. Status=<nil> <nil>
│
